nomad/state/state_store.go

```diff
@@ -2124,6 +2124,22 @@ func (s *StateStore) getJobStatus(txn *memdb.Txn, job *structs.Job, evalDelete b
 		}
 	}
 
+	// system jobs are running until explicitly stopped (which is handled elsewhere)
+	if job.Type == structs.JobTypeSystem {
+		if job.Stop {
+			return structs.JobStatusDead, nil
+		}
+
+		if hasEval {
```

Comment:
Lets just remove this and return running after checking if it has been explicitly stopped. The reason is that there could be no allocations (because no eligible nodes) and no evals because they have all garbage collected.